cmd/backend_kv_server.cpp

```diff
@@ -175,10 +177,12 @@ int main(int argc, char* argv[]) {
             [] { return std::make_unique<DummyServerCompletionQueue>(); },
         };
 
-        // remote sentry client
-        auto sentry_client = std::make_shared<silkworm::sentry::rpc::client::SentryClient>(
+        auto remote_sentry_client = std::make_shared<silkworm::sentry::rpc::client::SentryClient>(
             node_settings.external_sentry_addr,
             *context_pool.next_context().client_grpc_context());
+        auto sentry_client = std::make_shared<silkworm::sentry::SessionSentryClient>(
+            std::move(remote_sentry_client),
+            db::EthStatusDataProvider(db::ROAccess(database_env), node_settings.chain_config.value()).to_factory_function());
```

Comment:
I'm not a huge fan of these names: `EthStatusDataProvider` and `to_factory_function` used in this way aren't as intuitive as I wish at a first read.

However, any alternative that comes to my mind (i.e. `EthStatusSource` and `data_factory`) is not very good as well, so we need to think it a bit more.